### PR TITLE
fix!: privatize deprecated constructor

### PR DESCRIPTION
**Pull request by elharo**
@kolea2  one of last two deprecated methods to clean up before releasing 1.0

see https://github.com/googleapis/google-auth-library-java/pull/238

Can we take care of this now? I think https://www.github.com/googleapis/google-auth-library-java/issues/350 means we can do this now.

### CODE REVIEW

**Comment by elharo** [comment]
[ERROR] 7009: com.google.auth.oauth2.ServiceAccountJwtAccessCredentials: Accessibility of method 'public ServiceAccountJwtAccessCredentials(java.lang.String, java.lang.String, java.security.PrivateKey, java.lang.String)' has been decreased from public to private

is expected and this library is pre-1.0

File: clirr-ignored-differences.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- see https://www.mojohaus.org/clirr-maven-plugin/examples/ignored-differences.html -->
+<differences>
+  <difference>
+    <differenceType>7009</differenceType>
+    <className>com/google/auth/oauth2/ServiceAccountJwtAccessCredentials</className>
+    
+    com.google.auth.oauth2.ServiceAccountJwtAccessCredentials <<<<
+
```
Comment (chingor13) [inline clirr-ignored-differences.xml:9]: This looks like a merge error?

File: pom.xml [hunk 2]
```diff
@@ -305,6 +305,7 @@
         <configuration>
           <ignoredDifferencesFile>clirr-ignored-differences.xml</ignoredDifferencesFile>
           <logResults>true</logResults>
+          <failOnError>false</failOnError>
```
Comment (chingor13) [inline pom.xml:308]: I think we want to keep having it fail on non-ignored breaking changes.

**Review by chingor13 — CHANGES_REQUESTED** [review]

File: clirr-ignored-differences.xml (hunk 1 above)
Comment (elharo) [reply clirr-ignored-differences.xml]: fixed

File: pom.xml (hunk 2 above)
Comment (elharo) [reply pom.xml:308]: This is pre-1.0

**Review by chingor13 — APPROVED** [review]

**Pull request merged**